### PR TITLE
fix coreid not unique

### PR DESCRIPTION
**Pull request by jfbai**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

In many real world cases, many cores have same `CoreID`, in this case, function `CPUsInCores` and `isCoreFree` will return wrong error. For example, `CPUsInCores(0).ToSlice()` will return `[]int{0, 1, 8, 9}`.

```
	topoDualSocketHT = &topology.CPUTopology{
		NumCPUs:    16,
		NumSockets: 2,
		NumCores:   8,
		CPUDetails: map[int]topology.CPUInfo{
			0:  {CoreID: 0, SocketID: 0, NUMANodeID: 0},
			1:  {CoreID: 0, SocketID: 0, NUMANodeID: 0},
			2:  {CoreID: 1, SocketID: 0, NUMANodeID: 0},
			3:  {CoreID: 1, SocketID: 0, NUMANodeID: 0},
			4:  {CoreID: 2, SocketID: 0, NUMANodeID: 0},
			5:  {CoreID: 2, SocketID: 0, NUMANodeID: 0},
			6:  {CoreID: 3, SocketID: 0, NUMANodeID: 0},
			7:  {CoreID: 3, SocketID: 0, NUMANodeID: 0},
			8:  {CoreID: 0, SocketID: 1, NUMANodeID: 1},
			9:  {CoreID: 0, SocketID: 1, NUMANodeID: 1},
			10: {CoreID: 1, SocketID: 1, NUMANodeID: 1},
			11: {CoreID: 1, SocketID: 1, NUMANodeID: 1},
			12: {CoreID: 2, SocketID: 1, NUMANodeID: 1},
			13: {CoreID: 2, SocketID: 1, NUMANodeID: 1},
			14: {CoreID: 3, SocketID: 1, NUMANodeID: 1},
			15: {CoreID: 3, SocketID: 1, NUMANodeID: 1},
		},
	}
```

test case in `cpu_assignment_test.go` will fail

```
		{
			"take a socket of cpus from dual socket with HT",
			topoDualSocketHT16C,
			cpuset.NewCPUSet(0, 2, 4, 5, 6, 7, 8, 9, 10, 11),
			2,
			"",
			cpuset.NewCPUSet(4, 5),
		},
```

In this PR, we will masquerade `CoreID` in order to make `CoreID`s unique.

/sig node
/assign @klueska

### CODE REVIEW

**Comment by ehashman** [comment]
Please add an appropriate release note describing the bugfix in a ```release-note block in your PR description.

/priority important-soon
/triage accepted
/cc @fromanirh

**Review by ffromani — COMMENTED** [review]
Thanks for your PR! I'm adding some initial questions to better understand the context.
Is this PR fixing a issue you encountered? could you please file/link the issue in order to have more details?

File: pkg/kubelet/cm/cpumanager/topology/topology.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ func (topo *CPUTopology) CPUsPerSocket() int {
 }
 
 // CPUInfo contains the NUMA, socket, and core IDs associated with a CPU.
+// In some real world cases, CoreIDs on different sockets are the same,
```
[the remark this answers is not in the document]
Comment (jfbai) [reply pkg/kubelet/cm/cpumanager/topology/topology.go:64]: I see, I missed the `getUniqueCoreID` function, thanks a lot.

**Comment by jfbai** [comment]
> Thanks for your PR! I'm adding some initial questions to better understand the context.
> Is this PR fixing a issue you encountered? could you please file/link the issue in order to have more details?

We are doing implementation about an internal cpu policy and use our machines' topology as test case in `policy_test.go`, which is raw topology info and not generated from `Discover` function. Then I misunderstood the `CPUTopology`.

**Pull request closed without merging**